Add concepts section to docs. (#2050)
awslabs/gluon-ts: docs/concepts.md

# Concepts

## Estimator and Predictor

GluonTS uses two main abstractions called `Estimator` and `Predictor`.

Each `Predictor` implements `.predict(..)`, which given some input time-series
will return forecasts:

```py
forecasts = predictor.predict(data)
```

In contrast, an `Estimator` is trained to produce a `Predictor` which then is
used to make the actual predictions:

```py
predictor = estimator.train(train_data)
forecasts = predictor.predict(data)
```

The reason to split `Estimator` and `Predictor` into two classes is that many
models require a dedicated training step to generate a global model. This
global model is only trained once, but is used to make predictions for all

> time-serires.

Reviewer: ```suggestion
time-series.
```

This is in contrast to local models, which are fitted on individual time-series
and therefore try to capture the characteristics of each time-series but not
the dataset in its entirety.

Training a global model usually takes a lot of time: often hours, but sometimes
even days. Thus, it is not feasable to train the model "offline" (on demand).

> In contrast, a local model is usually fitted "oline" as part of the request.

Reviewer: ```suggestion
Training a global model can take a lot of time: up to hours, but sometimes
even days. Thus, it is not feasible to train the model "offline" (on demand).
In contrast, a local model is usually fitted "online" as part of the prediction.
```

However, I don't understand what this sentence means:

```
Thus, it is not feasible to train the model "offline" (on demand)
```

Author: The second part of the sentence is missing. It's not feasible to do it for each prediction but needs to do be done separately.

Reviewer: Do you want to add that?

Author: Yes

> In GluonTS models that are local are available as predictors, whilst global

Reviewer: ```suggestion
In GluonTS, local models are directly available as predictors, whilst global
```

objects has columns (or fields) which represent attributes of the
time-series.

> Most odels use the `target`-column to indicate the time-series that we want to

Reviewer: ```suggestion
Most models use the `target`-column to indicate the time-series that we want to
```
